apps/examples/smartfs/buffer_optimization/buffer_optimization_main.c

```diff
@@ -0,0 +1,449 @@
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ *
+ * apps/examples/smartfs/buffer_optimization/buffer_optimization_main.c
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/stat.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <time.h>
+
+#include <tinyara/timer.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+#define SAMPLE_FILE "/mnt/sample_file"
+#define TIMER_DEVNAME "/dev/timer%d"
+#define READ_TEST_ITR 100	//No. of times file will be read for testing
+#define APPEND_TEST_ITR 1	//No. of times file will be appended for testing
+#define TOTAL_APPEND_SIZE 10240	//Amount of data to be appended during the test
+
+/****************************************************************************
+ * Private data
+ ****************************************************************************/
+char g_buf[1025];
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: print_test_result
+ *
+ * Description: Helper function to print the test results.
+ *
+ ****************************************************************************/
+
+void print_test_result(const char *name, int size, int itr, int buf_size, uint32_t time)
+{
+	printf("\n**************************************************************************************\n");
+	printf("Throughput while %s %dKb file %d time(s) with %dbyte buffer size = %.02fKbps\n",\
+			name, size, itr, buf_size, (float)(size * 1000000) / time);
+	printf("\n**************************************************************************************\n");
+	return;
+}
+
+/****************************************************************************
+ * Name: create_test_file
+ *
+ * Description: Creates the test file with test data that we will use
+ *              to conduct the test.
+ *
+ ****************************************************************************/
+
+static int create_test_file(int f_size)
+{
+	int fd;
+	int ret;
+	memset(g_buf, 1024, '1');
+
+	fd = open(SAMPLE_FILE, O_WROK | O_CREAT);
+	if (fd < 0) {
+		printf("Unable to open Sample File for testing, Error = %d\n", errno);
+		return ERROR;
+	}
+
+	while (f_size >= 1024) {
```

Comment:
The sector size of 1024 bytes may be valid only with some chipsets. Why don't you get the current sector size and use it here?

Reply:
I missed one assumption here. If we add create_test_until(), then we need to use variable (i.e., optimized) buffer sizes.